x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/common/http/HttpClientTests.java

```diff
@@ -770,6 +770,50 @@ public void testCreateUri() throws Exception {
         assertCreateUri("https://example.org", "");
     }
 
+    public void testConnectionReuse() throws Exception {
```

Comment:
This was the least flaky way I could think of to test this but it's not exactly watertight, we cannot 100% guarantee that the connection will be re-used in the first case, nor can we guarantee that the client port won't be re-used in the second case.

I also couldn't work out a way to test that TCP keepalives are enabled at all.